grr/server/grr_response_server/gui/ui/lib/api_translation/client.ts

```diff
@@ -30,6 +32,124 @@ function createClientLabel(label: ApiClientLabel): ClientLabel {
   return {owner: label.owner, name: label.name};
 }
 
+function createAgentInfo(apiAgentInfo: ApiClientInformation): AgentInfo {
+  return {
+    clientName: apiAgentInfo.clientName,
+    clientVersion: apiAgentInfo.clientVersion,
+    revision: apiAgentInfo.revision,
+    buildTime: apiAgentInfo.buildTime,
+    clientBinaryName: apiAgentInfo.clientBinaryName,
+    clientDescription: apiAgentInfo.clientDescription,
+  }
+}
+
+function createOsInfo(apiUname: ApiUname): OsInfo {
+  return {
+    system: apiUname.system,
+    node: apiUname.node,
+    release: apiUname.release,
+    version: apiUname.version,
+    machine: apiUname.machine,
+    kernel: apiUname.kernel,
+    fqdn: apiUname.fqdn,
+    installDate: createOptionalDate(apiUname.installDate),
+    libcVer: apiUname.libcVer,
+    architecture: apiUname.architecture,
+  }
+}
+
+function createUser(apiUser: ApiUser): User {
+  return {
+    username: apiUser.username,
+    fullName: apiUser.fullName,
+    lastLogon: createOptionalDate(apiUser.lastLogon),
+    uid: apiUser.uid,
+    gid: apiUser.gid,
+    shell: apiUser.shell,
+    homedir: apiUser.homedir,
+  }
+}
+
+function createNetworkAddress(apiNetAddress: ApiNetworkAddress): NetworkAddress {
+  if (!apiNetAddress.addressType) throw new Error('addressType attribute is missing.');
+  if (!apiNetAddress.packedBytes) throw new Error('packedBytes attribute is missing.');
+
+  let addressType = 'IPv4'
+  if (apiNetAddress.addressType === 'INET6') {
+    addressType = 'IPv6';
+  }
+
+  const addressBytes = decodeBase64(apiNetAddress.packedBytes);
+
+  return {
+    addressType: addressType,
+    ipAddress: (addressType === 'IPv4') ? createIpv4Address(addressBytes) : createIpv6Address(addressBytes),
+  }
+}
+
+function createNetworkInterface(apiInterface: ApiInterface): NetworkInterface {
+  if (!apiInterface.macAddress) throw new Error('macAddress attribute is missing.');
+  if (!apiInterface.ifname) throw new Error('ifname attribute is missing.');
+
+  return {
+    macAddress: createMacAddress(decodeBase64(apiInterface.macAddress)),
+    interfaceName: apiInterface.ifname,
+    addresses: (apiInterface.addresses || []).map(createNetworkAddress),
```

Comment:
Use `??` instead of `||` when right-hand side should be a replacement for null-ish values (here and many more occurences).